src/session.ts

```diff
@@ -66,23 +71,46 @@ async function encryptSession(session: Session) {
   return sealData(session, { password: WORKOS_COOKIE_PASSWORD });
 }
 
-async function withAuth(
-  request: Request,
-  options?: {
-    ensureSignedIn?: false;
-    debug?: boolean;
-  },
-): Promise<UserInfo | NoUserInfo>;
-
-async function withAuth(
-  request: Request,
-  options: {
-    ensureSignedIn?: true;
-    debug?: boolean;
-  },
-): Promise<UserInfo>;
-
-async function withAuth(request: Request, { ensureSignedIn = false, debug = false } = {}) {
+type LoaderValue<Data> = Response | TypedResponse<Data> | NonNullable<Data> | null;
+type LoaderReturnValue<Data> = Promise<LoaderValue<Data>> | LoaderValue<Data>;
+
+type AuthLoader<Data> = (
+  args: LoaderFunctionArgs & { auth: AuthorizedData | UnauthorizedData },
+) => LoaderReturnValue<Data>;
+
+type AuthorizedAuthLoader<Data> = (args: LoaderFunctionArgs & { auth: AuthorizedData }) => LoaderReturnValue<Data>;
+
+export async function authkitLoader(
```

Comment:
Any reason why we're exporting all these overloads over just exporting `authkitLoader` at the end of this file?

Reply:
Ah not really, I was playing with this in a separate Remix app and just copy/pasted. Happy to revert.